### ISSUE TITLE
Sequence packing for SFT stage

### ISSUE THREAD

**Issue by trolommonm** [Victorwz/Open-Qwen2VL#19]
Why did you not use sequence packing in the SFT stage?

**Comment by Victorwz** [comment]
I recommend you to do so now. Actually, when I am working on this work, I did not have additional computes to well-tune the hyperparameters for the sft stage on packed sft dataset. The lr and number of epochs will be different between packed and unpacked sft data. This is why I did not do that. But I believe performing packing on SFT data will significantly enhance both the efficiency and efficacy.

**Comment by trolommonm** [comment]
Did you notice any difference in the training dynamics? With sequence packing, less optimization steps are taken. Have you guys done any experiments on how this may affect the training convergence? For e.g. with sample packing, do you notice that your final loss is higher than without sample packing?
